utopia-remix/app/util/api.server.ts

```diff
@@ -35,25 +36,47 @@ interface HandleRequest {
   params: Params<string>
 }
 
+export type ProjectAccessValidator = (request: Request, params: Params<string>) => Promise<boolean>
```

Comment:
given how they are used, there's not a reason for returning a value - either they succeed or not, and when failing the chain stops. Therefore either this returns a "true" value that is handled explicitly (e.g. throwing an exception) or it does not return anything because it will implicitly explode with e.g. `ensure` (or any other exception)